cupy/_core/dlpack.pyx

```diff
@@ -388,7 +485,15 @@ cpdef from_dlpack(array):
             stream = stream_module.get_current_stream_ptr()
             if stream == 0:
                 stream = stream_module.get_default_stream_ptr()
-            dltensor = array.__dlpack__(stream=stream)
+
+            # For backwards compatibility we catch TypeErrors for now.
+            try:
+                dltensor = array.__dlpack__(
+                    stream=stream, max_version=(1, 0), copy=copy)
```

Comment:
dito, re: https://github.com/cupy/cupy/pull/8683/files#r1809755389 (hard-wiring the `max_version`)